lib/src/Tabs/Tabs.tsx

```diff
@@ -1,22 +1,62 @@
 import React from "react";
 import { TabItemProps } from "./TabItem";
 import classnames from "classnames";
+import { useCombinedRefs } from "../hooks/useCombinedRef";
 
 export type TabsProps = JSX.IntrinsicElements["ul"] & {
     /** Index of focsued tab */
     value?: number;
+    /** set focused index to active index on focus */
+    autoSelectOnFocus?: boolean;
     /** Callback on tab item changed */
     onTabChange?: (index: number) => void;
+    /** Callback on tab item when delete key is triggered */
+    onTabDelete?: (index: number) => void;
 };
 /** Tabs organize and allow navigation between groups of content that are related and at the same level of hierarchy. */
-export const Tabs: React.FC<TabsProps> = React.forwardRef(({ value, onTabChange, ...props }: TabsProps, ref: React.ForwardedRef<HTMLUListElement>) => {
+export const Tabs: React.FC<TabsProps> = React.forwardRef(({ value, onTabChange, autoSelectOnFocus, onTabDelete, ...props }: TabsProps, ref: React.ForwardedRef<HTMLUListElement>) => {
+    const tabsRef: React.MutableRefObject<HTMLUListElement> = useCombinedRefs(ref);
+    const [focusedIndex, setFocusedIndex] = React.useState<number>(value);
+
+    const navigateToTab = (target: HTMLAnchorElement, event: React.MouseEvent | React.KeyboardEvent) => {
+        const nextFocusedIndex: number = parseInt(target.dataset.indexNumber);
+        setFocusedIndex(nextFocusedIndex);
+        target.href?.endsWith("#") && event.preventDefault();
+        if (target.dataset.disabled !== "true") {
+            onTabChange && onTabChange(nextFocusedIndex);
+        }
+    };
     const onClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
-        event.currentTarget.href.endsWith("#") && event.preventDefault();
-        onTabChange && onTabChange(parseInt(event.currentTarget.dataset.indexNumber));
+        navigateToTab(event.currentTarget, event);
+    };
+
+    const onKeyDown = (event: React.KeyboardEvent<HTMLUListElement>) => {
+        const items: HTMLAnchorElement[] = Array.from(tabsRef.current.querySelectorAll(".nav-tabs > .rc.nav-item > .nav-link"));
+        switch (event.key) {
+            case "Enter":
+            case " ":
+                navigateToTab(event.target as HTMLAnchorElement, event);
+                break;
+            case "ArrowLeft":
+            case "ArrowRight":
+                const direction: number = event.key === "ArrowLeft" ? -1 : 1;
+                const nextFocusedIndex: number = focusedIndex + direction;
```

Comment:
we can modify the algorithm slightly to support infinite moving (move to last when on first; move to first when on last)

```
const nextFocusedIndex: number = (focusedIndex + items.length + direction) % items.length;
```